packages/dd-trace/src/llmobs/sdk.js

```diff
@@ -0,0 +1,448 @@
+      return
+    }
+
+    if (!LLMObsTagger.tagMap.has(span)) {
+      logger.warn('Span must be an LLMObs-generated span')
+      return
+    }
+
+    try {
+      return {
+        traceId: span.context().toTraceId(true),
+        spanId: span.context().toSpanId()
+      }
+    } catch {
+      logger.warn('Faild to export span. Span must be a valid Span object.')
+      return undefined
+    }
+  }
+
+  submitEvaluation (llmobsSpanContext, options = {}) {
+    if (!this.enabled) {
+      logger.warn(
+        'LLMObs.submitEvaluation() called when LLMObs is not enabled. Evaluation metric data will not be sent.'
+      )
+      return
+    }
+
+    if (!this._config.llmobs.apiKey && !this._config.apiKey) {
+      logger.warn(
+        'DD_API_KEY is required for sending evaluation metrics. Evaluation metric data will not be sent.\n' +
+        'Ensure this configuration is set before running your application.'
+      )
+      return
+    }
+
+    const { traceId, spanId } = llmobsSpanContext
+    if (!traceId || !spanId) {
+      logger.warn(
+        'spanId and traceId must both be specified for the given evaluation metric to be submitted.'
+      )
+      return
+    }
+
+    const mlApp = options.mlApp || this._config.llmobs.mlApp
+    if (!mlApp) {
+      logger.warn('ML App name is required for sending evaluation metrics. Evaluation metric data will not be sent.')
+      return
+    }
+
+    const timestampMs = options.timestampMs || Date.now()
+    if (typeof timestampMs !== 'number' || timestampMs < 0) {
+      logger.warn('timestampMs must be a non-negative integer. Evaluation metric data will not be sent')
+      return
+    }
+
+    const { label, value, tags } = options
+    const metricType = options.metricType?.toLowerCase()
+    if (!label) {
+      logger.warn('label must be the specified name of the evaluation metric')
+      return
+    }
+    if (!metricType || !['categorical', 'score'].includes(metricType)) {
+      logger.warn('metricType must be one of "categorical" or "score"')
+      return
+    }
+
+    if (metricType === 'categorical' && typeof value !== 'string') {
+      logger.warn('value must be a string for a categorical metric.')
+      return
+    }
+    if (metricType === 'score' && typeof value !== 'number') {
+      logger.warn('value must be a number for a score metric.')
+      return
+    }
+
+    const evaluationTags = {
+      'dd-trace.version': tracerVersion,
+      ml_app: mlApp
+    }
+
+    if (tags) {
+      for (const key in tags) {
+        const tag = tags[key]
+        if (typeof tag === 'string') {
+          evaluationTags[key] = tag
+        } else if (typeof tag.toString === 'function') {
+          evaluationTags[key] = tag.toString()
+        } else {
+          logger.warn('Failed to parse tags. Tags for evaluation metrics must be strings')
+        }
+      }
+    }
+
+    const payload = {
+      span_id: spanId,
+      trace_id: traceId,
+      label,
```

Comment:
do we need to provide the field name here for consistency?

Reply:
i might be misunderstanding your question, but when passing a variable to an object like this in JS, it actually names the key the name of the variable, so if we have:
```javascript
const label = 'foo'
const payload = { label }
console.log(payload) // { label: 'foo' }
console.log(JSON.stringify(payload)) // { "label": "foo" }
```